### PR TITLE
CUMULUS-2578: Update an execution via API

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-2578: As a user, I want to update an execution](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2578)

## Changes

* Adds `PUT /executions` endpoint to update an execution
* cumulus-api document [PR](https://github.com/nasa/cumulus-api/pull/309)

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

File: packages/api/tests/endpoints/test-executions.js [hunk 1]
```diff
@@ -1002,3 +1002,221 @@ test.serial('POST /executions with non-existing parentArn still creates a new ex
...
+  const fetchedPgRecord = await executionPgModel.get(
+    t.context.knex,
+    {
+      arn: updatedExecution.arn,
+    }
+  );
+
+  t.is(fetchedPgRecord.arn, fetchedDynamoRecord.arn);
+  t.falsy(fetchedPgRecord.parent_cumulus_id);
+});
```
Comment (jaortega527) [inline packages/api/tests/endpoints/test-executions.js:1222]: I'm sure it's just my ignorance I see the way api/endpoints/executions.js/update() doesn't check for the existence of the parentArn/parent_cumulus_id and how that results in the update not failing for non-existent parentArn, but shouldn't it fail? Again, probably just my ignorance.

Comment (jennyhliu) [reply packages/api/tests/endpoints/test-executions.js:1222]: The current baseline ignores the parentArn is it doesn't exist.
https://github.com/nasa/cumulus/blob/master/packages/db/src/translate/executions.ts#L139

Comment (jaortega527) [reply packages/api/tests/endpoints/test-executions.js:1222]: Ah. Understood.

**Comment by jaortega527** [comment]
Looks great. Excellent tests.

**Review by jaortega527 — APPROVED** [review]
Looks great. Excellent tests.

**Pull request merged**